### PR TITLE
Removed torch7 models

### PR DESCRIPTION
**Pull request by alexlyulkov**
Merge wtih https://github.com/opencv/opencv/pull/24294

Removed torch7 models, because torch7 will not be supported in OpenCV 5:

removed Openface face recognition .t7 checkpoint
removed ENet image segmentation .net checkpoint
replaced Fast Neutral Style .t7 checkpoint by .onnx checkpoint

(this PR is a copy of  the closed PR #1096  but with a proper branch name)

### CODE REVIEW

**Comment by dkurt** [comment]
Please also remove the following:

* https://github.com/opencv/opencv_extra/tree/5.x/testdata/dnn/torch
* https://github.com/opencv/opencv_extra/blob/5.x/testdata/dnn/lena_feathers.png
* https://github.com/opencv/opencv_extra/blob/5.x/testdata/dnn/lena_starry_night.png

**Review by dkurt — CHANGES_REQUESTED** [review]
@alexlyulkov, please remove monster-size https://github.com/alexlyulkov/opencv_extra/blob/al/remove-torch7-from-dnn/testdata/dnn/torch_enet_prob.npy and missed https://github.com/alexlyulkov/opencv_extra/blob/al/remove-torch7-from-dnn/testdata/dnn/generate_torch_models.py

**Comment by asmorkalov** [comment]
@alexlyulkov Friendly reminder.

File: testdata/dnn/download_models.py [hunk 1]
```diff
@@ -260,14 +250,9 @@ def get_confirm_token(response):  # in case of large files
...
-        sha='5b5e115253197b84d6c6ece1dafe6c15d7105ca6',
-        filename='fast_neural_style_eccv16_starry_night.t7'),
-    Model(
-        name='Fast-Neural-Style',
-        url='http://cs.stanford.edu/people/jcjohns/fast-neural-style/models/instance_norm/feathers.t7',
-        sha='9838007df750d483b5b5e90b92d76e8ada5a31c0',
-        filename='fast_neural_style_instance_norm_feathers.t7'),
+        url='https://media.githubusercontent.com/media/onnx/models/main/vision/style_transfer/fast_neural_style/model/mosaic-9.onnx',
+        sha='dc3ef19f7f8a5d906c97c84cd4dd2732bb73960a',
+        filename='mosaic-9.onnx'),
```
Comment (opencv-alalek) [inline testdata/dnn/download_models.py:255]: ONNX models should go to `onnx/models/`

**Review by dkurt — APPROVED** [review]

**Pull request merged**